public/js/reducers/tabs.js

```diff
@@ -47,4 +60,5 @@ function getTabId(tab) {
   return id;
 }
 
+const update = setupReducer(initialState, { addTabs, selectTab });
```

Comment:
It's a good question and comes up a lot in the redux community. However, I think most people end up just using case statements after getting used to it. A few counter-points (sorry, this got a bit long but it's a topic that always comes up in redux so I wanted to be complete):

1. Using separate functions encourages doing more work in a single reducer. A reducer shouldn't get too big, and if it does should be broken into subreducers. The breakpoint and sources reducers are about as big as they should be, which is ~110 lines. The case statements are very easy to manage at that level, it's not really a large function.
2. It is not the same as a large functions scattered across the codebase. The core design of redux works this way and handling multiple actions in a function is good, not a bad design.
3. Separating out functions makes it visually indistinguishable from actions. It's nice to clearly see that you're working in a reducer. You end up having two `addBreakpoint` functions, one action and one reducer. If we do this, we need to use something like tofino's `set` prefix for all the functions for clarity.
4. We definitely want action types to use THIS_FORMAT, but your followup comment points out another technique that still keeps this.
5. By the time you use something like `createReducer`, I think there's more syntax than a simple switch statements, and I'm not sure I see the benefit.
6. You *never* want to test individual reducer handlers. Always just pass state into the reducer itself.
7. The only reason tofino has those functions in that file is because they are called many times by different actions. We do the same thing with `updateText` and other helper functions in the sources reducer. So the pattern is that you pull out things that need to be used in multiple places; tofino just happens to need to do anything else in that reducer.
8. In my opinion, what's more important is what the console is doing, and they use raw switch statements:

https://github.com/devtools-html/gecko-dev/blob/a09e19c97e761b4cf69be92e9ebc40cd6aa8be4d/devtools/client/webconsole/new-console-output/reducers/messages.js

https://github.com/devtools-html/gecko-dev/blob/a09e19c97e761b4cf69be92e9ebc40cd6aa8be4d/devtools/client/webconsole/new-console-output/reducers/filters.js

Especially with smaller reducers with many action types, I think the `switch` statement is a clear winner. We could probably clean up/break up the sources & breakpoints reducer a bit because some of the case handlers are a little large. But I don't want to enforce moving all of the code out of switch statements.

Most importantly, I don't want to merge anything that only changes one reducer. If we do anything, it needs to be done to all reducers at once. We already have other things that are half-implemented, and I don't want more inconsistency within the code.

Let's hold off on this until after the dust settles for turning on the debugger. I'd like to focus on that for the next few days.